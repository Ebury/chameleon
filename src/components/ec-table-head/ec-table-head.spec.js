import { mount } from '@vue/test-utils';
import EcTableHead from './ec-table-head.vue';

describe('EcTableHead', () => {
  it('Should render as expected', () => {
    const propsData = {
      columns: [
        {
          name: 'Column A',
        },
        {
          name: 'Column B',
          span: 2,
        },
        {
          name: 'Column C',
          scope: 'row',
        },
      ],
    };
    const wrapper = mount(EcTableHead, {
      propsData,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('Should not render if no columns are supplied', () => {
    const wrapper = mount(EcTableHead, {
      propsData: {},
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
