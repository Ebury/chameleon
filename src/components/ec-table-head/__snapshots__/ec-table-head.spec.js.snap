// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcTableHead should not render if no columns are supplied 1`] = `<!---->`;

exports[`EcTableHead should render as expected 1`] = `
<thead
  class="ec-table-head"
  data-test="ec-table-head"
>
  <tr>
    <th
      class="ec-table-head__cell"
      data-test="ec-table-head__cell ec-table-head__cell--0"
      scope="col"
    >
      <span
        class="ec-table-head__cell-wrapper"
      >
        <span>
          
          Column A
        
        </span>
         
        <!---->
         
        <!---->
      </span>
    </th>
    <th
      class="ec-table-head__cell"
      colspan="2"
      data-test="ec-table-head__cell ec-table-head__cell--1"
      scope="col"
    >
      <span
        class="ec-table-head__cell-wrapper"
      >
        <span>
          
          Column B
        
        </span>
         
        <!---->
         
        <!---->
      </span>
    </th>
    <th
      class="ec-table-head__cell"
      data-test="ec-table-head__cell ec-table-head__cell--2"
      scope="col"
    >
      <span
        class="ec-table-head__cell-wrapper"
      >
        <span>
          
          Column C
        
        </span>
         
        <!---->
         
        <!---->
      </span>
    </th>
  </tr>
</thead>
`;

exports[`EcTableHead should render info icon with a tooltip as expected 1`] = `
<thead
  class="ec-table-head"
  data-test="ec-table-head"
>
  <tr>
    <th
      class="ec-table-head__cell"
      data-test="ec-table-head__cell ec-table-head__cell--0"
      scope="col"
    >
      <span
        class="ec-table-head__cell-wrapper"
      >
        <span>
          
          Column A
        
        </span>
         
        <svg
          class="ec-table-head__info-icon ec-table-head__icon ec-icon ec-icon--interactive ec-has-tooltip"
          data-original-title="null"
          height="16"
          width="16"
        >
          <use
            xlink:href="#ec-simple-info"
          />
        </svg>
         
        <!---->
      </span>
    </th>
    <th
      class="ec-table-head__cell"
      colspan="2"
      data-test="ec-table-head__cell ec-table-head__cell--1"
      scope="col"
    >
      <span
        class="ec-table-head__cell-wrapper"
      >
        <span>
          
          Column B
        
        </span>
         
        <svg
          class="ec-table-head__info-icon ec-table-head__icon ec-icon ec-icon--interactive ec-has-tooltip"
          data-original-title="null"
          height="16"
          width="16"
        >
          <use
            xlink:href="#ec-simple-info"
          />
        </svg>
         
        <!---->
      </span>
    </th>
    <th
      class="ec-table-head__cell"
      data-test="ec-table-head__cell ec-table-head__cell--2"
      scope="col"
    >
      <span
        class="ec-table-head__cell-wrapper"
      >
        <span>
          
          Column C
        
        </span>
         
        <svg
          class="ec-table-head__info-icon ec-table-head__icon ec-icon ec-icon--interactive ec-has-tooltip"
          data-original-title="null"
          height="16"
          width="16"
        >
          <use
            xlink:href="#ec-simple-info"
          />
        </svg>
         
        <!---->
      </span>
    </th>
  </tr>
</thead>
`;

exports[`EcTableHead should render sorting as expected 1`] = `
<thead
  class="ec-table-head"
  data-test="ec-table-head"
>
  <tr>
    <th
      class="ec-table-head__cell"
      data-test="ec-table-head__cell ec-table-head__cell--0"
      scope="col"
    >
      <span
        class="ec-table-head__cell-wrapper"
      >
        <span>
          
          Column A
        
        </span>
         
        <!---->
         
        <a
          class="ec-table-sort ec-table-head__icon"
          data-test="ec-table-sort"
          href="#"
          title="Sorted ascending"
        >
          <svg
            class="ec-icon ec-table-sort__icon ec-table-sort__icon--asc"
            data-test="ec-table-sort__icon"
            height="16"
            width="16"
          >
            <use
              xlink:href="#ec-simple-arrow-drop-down"
            />
          </svg>
           
          <span
            class="ec-table-sort__text"
          >
            asc
          </span>
        </a>
      </span>
    </th>
    <th
      class="ec-table-head__cell"
      colspan="2"
      data-test="ec-table-head__cell ec-table-head__cell--1"
      scope="col"
    >
      <span
        class="ec-table-head__cell-wrapper"
      >
        <span>
          
          Column B
        
        </span>
         
        <!---->
         
        <a
          class="ec-table-sort ec-table-head__icon"
          data-test="ec-table-sort"
          href="#"
          title="Sorted descending"
        >
          <svg
            class="ec-icon ec-table-sort__icon ec-table-sort__icon--desc"
            data-test="ec-table-sort__icon"
            height="16"
            width="16"
          >
            <use
              xlink:href="#ec-simple-arrow-drop-down"
            />
          </svg>
           
          <span
            class="ec-table-sort__text"
          >
            desc
          </span>
        </a>
      </span>
    </th>
    <th
      class="ec-table-head__cell"
      data-test="ec-table-head__cell ec-table-head__cell--2"
      scope="col"
    >
      <span
        class="ec-table-head__cell-wrapper"
      >
        <span>
          
          Column C
        
        </span>
         
        <!---->
         
        <a
          class="ec-table-sort ec-table-head__icon"
          data-test="ec-table-sort"
          href="#"
          title="Not sorted"
        >
          <svg
            class="ec-icon ec-table-sort__icon"
            data-test="ec-table-sort__icon"
            height="16"
            width="16"
          >
            <use
              xlink:href="#ec-simple-arrow-up-down"
            />
          </svg>
           
          <!---->
        </a>
      </span>
    </th>
  </tr>
</thead>
`;
